Proposals/ReplaceTrustedAppsWithTags.md

```diff
@@ -0,0 +1,272 @@
+authorization; for instance when granting a new privilege to an app, users
+expect the app to be able to use the new privilege immediately.
+
+When App Authorization documents are presented while acquiring an OAuth2
+Bearer Token from a competent Authorization Server, it is expected that the
+authorization server will at least revalidate all authentication and authorization
+resources, to be responsive to changes in authentication and authorization
+properties.  The authorization server can influence the revalidation period
+by extending or diminishing the lifetime of the access tokens it issues. To
+mitigate an amplification or denial-of-service attack, the authorization
+server **SHOULD** provide for a minimum caching period (on the order of
+seconds) of all authentication and authorization resources it loads.  To
+respond immediately to normal changes in authorization, when App Authorizations
+are revalidated on each presentation, an app can simply forget its access
+tokens and acquire new ones.
+
+When App Authorization documents are presented in HTTP `Link` headers, the
+resource server **SHOULD** cache these authorization resources as normal, but
+[**SHOULD CONSIDER**](https://tools.ietf.org/html/rfc6919#section-2) reducing
+cache lifetimes to be more responsive to changes. Alternatively, permission
+management systems
+[**WOULD PROBABLY**](https://tools.ietf.org/html/rfc6919#section-5) use fresh
+URIs for updated App Authorization documents when possible; note however that
+the user's profile URI can't be changed for this purpose.
+
+Out of Scope
+============
+The method by which an app requests that the user authorize scopes/tags is
+beyond the scope of this memo. Methods could include, but are not limited to,
+an enumeration of desired tags in an app manifest file, posting a request to
+an inbox, new attributes on requests to an authentication server, and
+presentation of a request message to the user for manual editing of static
+files.
+
+The method by which apps are authorized for tags and modes is beyond the scope
+of this memo. Methods could include, but are not limited to, a privileged app
+that manages App Authorization documents and index files (which might read
+an inbox or otherwise receive requests from apps), a privileged app that the
+user uses to directly adjust the authorizations of apps, an authentication
+server that does double-duty as an authorization management system and which
+dynamically generates App Authorization documents, and manual editing of
+static files by the user.
+
+The method by which the tags that are in use by resources on resource servers
+can be discovered by apps or users is beyond the scope of this memo. Methods
+could include, but are not limited to, resource servers including tag-related
+information in HTTP `403` responses, resource servers posting events to an
+inbox, resource servers including human-readable information on a FAQ page,
+and well-known convention.
+
```

Comment:
I think we do need to address this concern. In many cases user which authorizes client doesn't act as [*resource controller*](https://github.com/solid/authentication-panel/blob/master/oidc-authentication.md#resource-controller). In that case user who has *read* (or even *read-write*) access to the resource would not be able to read associated ACL resource to know which tags they need to add to client authorization.

Reply:
agreed that this should be addressed. however, i don't believe it needs to be addressed in *this* document; it is an area for additional proposals.

Reply:
i think it's premature to try to design a technical solution to any of these three Out-of-Scope items right now. i think two parallel courses are likely initially:

1. a basic vocabulary of common Solid tags will emerge, and users will assign these basic tags for the default/global `"*"` origin to the various basic apps they use ;
2. users will either ask foreign owners/controllers what specialty/non-basic tag vocabulary they're using for their resources, or find them in a FAQ or "getting started" document, or (possibly) give apps the `"*"` tag for specific foreign origins where they don't know (or care) about the tag vocabulary being used, and don't mind that the app will be able to access all resources at that origin to which the user would otherwise have access.

giving an app the `"*"` tag for a specific foreign origin isn't ideal, but it's way better than giving it full access to every origin (especially the one(s) where your storage is). and note that giving tag `"*"` to origin `"*"` is specifically forbidden (see line 123).